### ISSUE TITLE
runtimeerror: given groups=1, weight of size [32, 128, 3, 3], expected input[1, 192, 160, 160] to have 128 channels, but got 192 channels instead 

### ISSUE THREAD

**Issue by AfsahS** [qubvel/segmentation_models.pytorch#250]
Hi, I am trying to implement Unet with ResNeSt backbone. Given architecture of Unet works fine with ResNeSt-50 but when  I try to to use ResNeSt-101, ResNeSt-200 or ResNeSt-269 it gives me following error 
runtimeerror: given groups=1, weight of size [32, 128, 3, 3], expected input[1, 192, 160, 160] to have 128 channels, but got 192 channels instead

Any idea how this can be solved?

**Comment by skyqwe123** [comment]
you can transpose the axis before forward(x)

**Comment by haigh1510** [comment]
@Afsah09 Hi! Try to set `out_channels=(3, 128, 256, 512, 1024, 2048)` for `resenest101` https://github.com/qubvel/segmentation_models.pytorch/blob/bdd5db25ae41d53eef1833df1a9e16fc141155d8/segmentation_models_pytorch/encoders/resnest.py#L73

**Comment by gg22mm** [comment]
Me too. How to solve this problem?

**Comment by gg22mm** [comment]
> 
> 
> Hi, I am trying to implement Unet with ResNeSt backbone. Given architecture of Unet works fine with ResNeSt-50 but when I try to to use ResNeSt-101, ResNeSt-200 or ResNeSt-269 it gives me following error
> runtimeerror: given groups=1, weight of size [32, 128, 3, 3], expected input[1, 192, 160, 160] to have 128 channels, but got 192 channels instead
> 
> Any idea how this can be solved?

# 数据增强
def get_training_augmentation():
    train_transform = [
        albu.HorizontalFlip(p=0.5),
        albu.ShiftScaleRotate(scale_limit=0.5, rotate_limit=0, shift_limit=0.1, p=1, border_mode=0),
        albu.PadIfNeeded(min_height=320, min_width=320, always_apply=True, border_mode=0),
        albu.RandomCrop(height=320, width=320, always_apply=True),
        albu.IAAAdditiveGaussianNoise(p=0.2),
        albu.IAAPerspective(p=0.5),
        albu.OneOf(
            [
                albu.CLAHE(p=1),
                albu.RandomBrightness(p=1),
                albu.RandomGamma(p=1),
            ],
            p=0.9,
        ),

        albu.OneOf(
            [
                albu.IAASharpen(p=1),
                albu.Blur(blur_limit=3, p=1),
                albu.MotionBlur(blur_limit=3, p=1),
            ],
            p=0.9,
        ),

        albu.OneOf(
            [
                albu.RandomContrast(p=1),
                albu.HueSaturationValue(p=1),
            ],
            p=0.9,
        ),
    ]
    return albu.Compose(train_transform)


def get_validation_augmentation():
    """Add paddings to make image shape divisible by 32"""
    test_transform = [
        albu.PadIfNeeded(384, 480)
    ]
    return albu.Compose(test_transform)


def to_tensor(x, **kwargs):
    return x.transpose(2, 0, 1).astype('float32')


def get_preprocessing(preprocessing_fn):
    """Construct preprocessing transform
    
    Args:
        preprocessing_fn (callbale): data normalization function 
            (can be specific for each pretrained neural network)
    Return:
        transform: albumentations.Compose
    
    """    
    _transform = [
        albu.Lambda(image=preprocessing_fn),
        albu.Lambda(image=to_tensor, mask=to_tensor),
    ]
    return albu.Compose(_transform)




#### Visualize resulted augmented images and masks

augmented_dataset = Dataset(
    x_train_dir, 
    y_train_dir, 
    augmentation=get_training_augmentation(), 
    classes=['car'],
)

# same image with different random transforms
for i in range(3):
    image, mask = augmented_dataset[1]
    visualize(image=image, mask=mask.squeeze(-1))

------------------- Picture enhancement must be run ---------------------

In [10]:

import albumentations as albu

In [11]:

def get_training_augmentation():
    train_transform = [

        albu.HorizontalFlip(p=0.5),

        albu.ShiftScaleRotate(scale_limit=0.5, rotate_limit=0, shift_limit=0.1, p=1, border_mode=0),

        albu.PadIfNeeded(min_height
